src/frontend/parser.mly

```diff
@@ -193,47 +207,51 @@ future_keyword:
 
 decl_identifier:
   | id=identifier { id }
-  | id=reserved_word { id }
+  | id_flag=reserved_word { reserved (fst id_flag) }
+
+decl_identifier_after_comma:
+  | id=identifier { id }
+  | id_flag=reserved_word { reserved_decl id_flag }
 
 reserved_word:
-  (* Keywords cannot be identifiers but
-     semantic check produces a better error message. *)
-  | FUNCTIONBLOCK { build_id "functions" $loc }
-  | DATABLOCK { build_id "data" $loc }
-  | PARAMETERSBLOCK { build_id "parameters" $loc }
-  | MODELBLOCK { build_id "model" $loc }
-  | RETURN { build_id "return" $loc }
-  | IF { build_id "if" $loc }
-  | ELSE { build_id "else" $loc }
-  | WHILE { build_id "while" $loc }
-  | FOR { build_id "for" $loc }
-  | IN { build_id "in" $loc }
-  | BREAK { build_id "break" $loc }
-  | CONTINUE { build_id "continue" $loc }
-  | VOID { build_id "void" $loc }
-  | INT { build_id "int" $loc }
-  | REAL { build_id "real" $loc }
-  | COMPLEX { build_id "complex" $loc }
-  | VECTOR { build_id "vector" $loc }
-  | ROWVECTOR { build_id "row_vector" $loc }
-  | MATRIX { build_id "matrix" $loc }
-  | COMPLEXVECTOR { build_id "complex_vector" $loc }
-  | COMPLEXROWVECTOR { build_id "complex_row_vector" $loc }
-  | COMPLEXMATRIX { build_id "complex_matrix" $loc }
-  | ORDERED { build_id "ordered" $loc }
-  | POSITIVEORDERED { build_id "positive_ordered" $loc }
-  | SIMPLEX { build_id "simplex" $loc }
-  | UNITVECTOR { build_id "unit_vector" $loc }
-  | CHOLESKYFACTORCORR { build_id "cholesky_factor_corr" $loc }
-  | CHOLESKYFACTORCOV { build_id "cholesky_factor_cov" $loc }
-  | CORRMATRIX { build_id "corr_matrix" $loc }
-  | COVMATRIX { build_id "cov_matrix" $loc }
-  | PRINT { build_id "print" $loc }
-  | REJECT { build_id "reject" $loc }
-  | TARGET { build_id "target" $loc }
-  | GETLP { build_id "get_lp" $loc }
-  | PROFILE { build_id "profile" $loc }
-  | TUPLE { build_id "tuple" $loc }
+  (* Keywords cannot be identifiers but it is nice to
+    let them parse as such to provide a better error *)
+  | FUNCTIONBLOCK { build_id "functions" $loc, false }
```

Comment:
I wonder if `build_id` is worth it when the `Ast.identifier` is destructured immediately. Alternative would just be the tuple `("functions", $loc, false)`.